homeassistant/components/elmax/coordinator.py

```diff
@@ -122,3 +122,53 @@ async def _async_update_data(self):
                 "Make sure the panel is online and that  "
                 "your firewall allows communication with it."
             ) from err
+
+        # Store a dictionary for fast endpoint state access
+        self._state_by_endpoint = {k.endpoint_id: k for k in status.all_endpoints}
+
+        # If panel supports it and a it hasn't been registered yet, register the push notification handler
+        if status.push_feature and self._push_notification_handler is None:
+            self._register_push_notification_handler()
+
+        self._fire_data_update(status)
+        return status
+
+    def _fire_data_update(self, status: PanelStatus):
+        # Store a dictionary for fast endpoint state access
+        self._state_by_endpoint = {k.endpoint_id: k for k in status.all_endpoints}
+        # Send the event data to every single device
+        for k, ep_status in self._state_by_endpoint.items():
+            event_signal = f"{SIGNAL_PANEL_UPDATE}-{self.panel_entry.hash}-{k}"
+            async_dispatcher_send(self.hass, event_signal, ep_status)
+
+        self.async_set_updated_data(status)
+
+    def _register_push_notification_handler(self):
+        ws_ep = (
+            f"{'wss' if self.http_client.base_url.scheme == 'https' else 'ws'}"
+            f"://{self.http_client.base_url.host}"
+            f":{self.http_client.base_url.port}"
+            f"{self.http_client.base_url.path}/push"
+        )
+        self._push_notification_handler = PushNotificationHandler(
+            endpoint=str(ws_ep),
+            http_client=self.http_client,
+            ssl_context=self.http_client.ssl_context,
+        )
+        self._push_notification_handler.register_push_notification_handler(
+            self._push_handler
+        )
+        self._push_notification_handler.start(loop=self.hass.loop)
+
+    async def _push_handler(self, status: PanelStatus) -> None:
+        self._fire_data_update(status)
+
+    async def async_shutdown(self) -> None:
```

Comment:
May want to call this when home assistant stop event happens on shutdown like #117830